File: superstitioMain/java/superstitio/cards/general/SkillCard/gainEnergy/ForceOrgasm.java
package superstitio.cards.general.SkillCard.gainEnergy;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.general.GeneralCard;
import superstitio.powers.SexualHeat;
import superstitioapi.actions.XCostAction;

public class ForceOrgasm extends GeneralCard {
    public static final String ID = DataManager.MakeTextID(ForceOrgasm.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.UNCOMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = -1;
    private static final int MAGIC = 15;
    private static final int UPGRADE_MAGIC = 5;
    private static final int ExhaustNum = 1;

    public ForceOrgasm() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
        this.exhaust = true;
//        ExhaustiveVariable.setBaseValue(this, 2);
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        int magicNumber = this.magicNumber;
        addToBot(new XCostAction(this, AbstractGameAction.ActionType.ENERGY, effect -> {
            SexualHeat.addToBot_addSexualHeat(AbstractDungeon.player, effect * magicNumber);
            addToBot_applyPower(new superstitio.powers.Overdraft(AbstractDungeon.player, effect * ExhaustNum));
        }));
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/general/SkillCard/gainEnergy/ForceOrgasm.kt
package superstitio.cards.general.SkillCard.gainEnergy

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.general.GeneralCard
import superstitio.powers.Overdraft
import superstitio.powers.SexualHeat
import superstitioapi.actions.XCostAction

class ForceOrgasm : GeneralCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
        this.exhaust = true
        //        ExhaustiveVariable.setBaseValue(this, 2);
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        val magicNumber = this.magicNumber
        addToBot(XCostAction(this, AbstractGameAction.ActionType.ENERGY) { effect: Int ->
            SexualHeat.addToBot_addSexualHeat(AbstractDungeon.player, effect * magicNumber)
            addToBot_applyPower(Overdraft(AbstractDungeon.player, effect * ExhaustNum))
        })
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(ForceOrgasm::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.UNCOMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = -1
        private const val MAGIC = 15
        private const val UPGRADE_MAGIC = 5
        private const val ExhaustNum = 1
    }
}
